internal/vault/cache_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"fmt"
+	"testing"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_clientCache_Prune(t *testing.T) {
+	dummyCallbackFunc := func(key, value interface{}) {}
+
+	tests := []struct {
+		name                  string
+		filterFuncReturnsTrue bool
+		cacheSize             int
+		cacheLen              int
+	}{
+		{
+			name:                  "filterFunc returns true",
+			filterFuncReturnsTrue: true,
+			cacheSize:             256,
+			cacheLen:              1,
+		},
+		{
+			name:                  "filterFunc returns false",
+			filterFuncReturnsTrue: false,
```

Comment:
Definitely not required, but it would be nice to include some filter function testing aside from the ones you have here. Perhaps one that does cache key prefix match, or something like that.

Reply:
This probably is easier to do once the Client interface is refactored. We'd need to use GetCacheKey() if we need to match since the filter is `func(Client) bool`.